src/overseer/core.clj

```diff
@@ -1,11 +1,88 @@
 (ns ^:no-doc overseer.core
   "Internal core functions"
   (:require [datomic.api :as d]
-            [clojure.set :as set]))
+            [clojure.set :as set]
+            [miner.herbert :as h]
+            loom.graph))
+
+(def job-schema
+  "Jobs are defined as maps with the requisite descriptive keys/values."
+  '{:job/id str
+    :job/type kw
+    :job/status (or :unstarted :started :finished :failed :aborted)
+    :job/heartbeat? int
```

Comment:
@elliot42 `:job/heartbeat` became optional which is important; jobs going into the store (which is when they're validated) don't have a heartbeat yet